### PR TITLE
ztp: Relocate the IBU source-cr

### PR DESCRIPTION
**Pull request by sakhoury**
The IBU source-cr is relocated to the ibu directory. The seed image reference is also commented out to allow for concise stage transitions. The `ManagedClusterSetBinding` name is modified to reflect the matching clusterSet.

`ztp/source-crs/ibu/ImageBasedUpgradeStatus.yaml` is added to allow for successful stage transition monitoring via PGT.

/cc @Missxiaoguo @imiller0

### CODE REVIEW

File: ztp/gitops-subscriptions/argocd/example/image-based-upgrades/example-oadp-policy.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: cluster.open-cluster-management.io/v1beta2
 kind: ManagedClusterSetBinding
 metadata:
-  name: ztp-common-mcsb
+  name: global
```
[the remark this answers is not in the document]
Comment (sakhoury) [reply ztp/gitops-subscriptions/argocd/example/image-based-upgrades/example-oadp-policy.yaml:4]: The `ManagedClusterSetBinding` must have the same name as the target managed `clusterSet`. The example strives to make use of the existing `global` cluster set. The user may create a new cluster set should they have overlapping use-cases.

**Comment by Missxiaoguo** [comment]
/lgtm

As we discussed, to ensure IBU policies accurately reflect the current upgrade status during the entire upgrade procedure, two policies are needed for each stage.
`*-stage-policy`  -> for stage transition
`*-status-policy` -> for stage status monitoring

ImageBasedUpgradeStatus CR which only contains status field helps to achieve that.

**Review by imiller0 — APPROVED** [review]
/lgtm

**Pull request merged**